shiny/ui/_navs_dynamic.py

```diff
@@ -0,0 +1,205 @@
+__all__ = (
+    "nav_insert",
+    "nav_remove",
+    "nav_hide",
+    "nav_show",
+)
+
+import sys
+from typing import Optional, Union
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+from .._docstring import add_example
+from .._namespaces import resolve_id
+from ..session import Session, require_active_session
+from ..types import NavSetArg
+from ._input_update import update_navs
+from ._navs import menu_string_as_nav
+
+
+@add_example()
+def nav_insert(
+    id: str,
+    nav_panel: Union[NavSetArg, str],
+    target: Optional[str] = None,
+    position: Literal["after", "before"] = "after",
+    select: bool = False,
+    session: Optional[Session] = None,
+) -> None:
+    """
+    Insert a new nav item into a navigation container.
+
+    Parameters
+    ----------
+    id
+        The ``id`` of the relevant navigation container (i.e., ``navset_*()`` object).
```

Comment:
At one point I think it was required to do double backtick to format code, but I don't think that's the case anymore

```suggestion
        The `id` of the relevant navigation container (i.e., `navset_*()` object).
```

Reply:
Ah cool, I'd made that change but not pushed yet, so good that it's correct! 😅